src/system/displays/displays_windows.cpp

```diff
@@ -21,44 +22,40 @@
-	std::vector<iware::system::display_t> ret;
+	const struct bundle {
+		HDC desktop_dc;
+		std::vector<iware::system::display_t> ret;
+	} bundle{GetDC(nullptr), {}};
+	iware::detail::quickscope_wrapper desktop_dc_deleter{[&]() { ReleaseDC(nullptr, bundle.desktop_dc); }};
 
 	EnumDisplayMonitors(
-	    nullptr, nullptr,
+	    bundle.desktop_dc, nullptr,
 	    [](auto, auto hdc, auto rect, auto userdata) {
-		    static const auto desktop_dc          = GetDC(nullptr);
-		    static const unsigned int desktop_dpi = GetDeviceCaps(desktop_dc, LOGPIXELSX);
-		    // https://blogs.msdn.microsoft.com/oldnewthing/20101013-00/?p=12543
-		    static const unsigned int desktop_bpp    = GetDeviceCaps(desktop_dc, BITSPIXEL) * GetDeviceCaps(desktop_dc, PLANES);
-		    static const double desktop_refresh_rate = GetDeviceCaps(desktop_dc, VREFRESH);
+		    auto& bundle = *reinterpret_cast<struct bundle*>(userdata);
 
+		    const unsigned int desktop_dpi = GetDeviceCaps(bundle.desktop_dc, LOGPIXELSX);
+		    // https://blogs.msdn.microsoft.com/oldnewthing/20101013-00/?p=12543
+		    const unsigned int desktop_bpp    = GetDeviceCaps(bundle.desktop_dc, BITSPIXEL) * GetDeviceCaps(bundle.desktop_dc, PLANES);
+		    const double desktop_refresh_rate = GetDeviceCaps(bundle.desktop_dc, VREFRESH);
 
-		    // https://msdn.microsoft.com/en-us/library/windows/desktop/dn280510(v=vs.85).aspx
-		    // Doesn't work, can't link to shcore
-		    //
-		    // unsigned int dpi_x;
-		    // unsigned int dpi_y;
-		    // GetDpiForMonitor(monitor /* arg #1 */, MDT_DEFAULT, &dpi_x, &dpi_y);
 
-		    // Sometimes returns 0 for some reason?
-		    // Fall back to the desktop's globals if so.
+		    // Sometimes returns 0 – fall back to the desktop's globals if so.
 		    const unsigned int monitor_dpi    = GetDeviceCaps(hdc, LOGPIXELSX);
 		    const unsigned int monitor_bpp    = GetDeviceCaps(hdc, BITSPIXEL) * GetDeviceCaps(hdc, PLANES);
 		    const double monitor_refresh_rate = GetDeviceCaps(hdc, VREFRESH);
 
 		    const unsigned int width  = std::abs(rect->right - rect->left);
 		    const unsigned int height = std::abs(rect->bottom - rect->top);
 
-		    auto& ret = *reinterpret_cast<std::vector<iware::system::display_t>*>(userdata);
 		    // See http://stackoverflow.com/a/12654433/2851815 and up for DPI. In short: can't be done too too well, go with best solution.
-		    ret.push_back({width, height, monitor_dpi ? monitor_dpi : desktop_dpi, monitor_bpp ? monitor_bpp : desktop_bpp,
-		                   monitor_refresh_rate ? monitor_refresh_rate : desktop_refresh_rate});
+		    bundle.ret.push_back({width, height, monitor_dpi ? monitor_dpi : desktop_dpi, monitor_bpp ? monitor_bpp : desktop_bpp,
+		                          monitor_refresh_rate ? monitor_refresh_rate : desktop_refresh_rate});
 
-		    return 1;
+		    return TRUE;
 	    },
-	    reinterpret_cast<LPARAM>(&ret));
+	    reinterpret_cast<LPARAM>(&bundle));
 
-	return ret;
+	return bundle.ret;
```

Comment:
@ThePhD do I need a std::move for this or is this handled by (M)NRVO even in a subobject?

Reply:
Sorry it took so long to get back here.

Guaranteed Copy Elision / Return Value Optimization does not apply to the suboject here: it will inflict a copy. I tested it with a `nocopy` object even on latest standard with the latest rules: no dice, it tries to copy (https://godbolt.org/z/Ge3GYqozh).

So, I think you'd have to `std::move(bundle.ret)` here.